sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -491,9 +491,13 @@ class SSLTransport(_AbstractTransport):
 
     def __init__(self, host, *, port=AMQPS_PORT, socket_timeout=None, ssl_opts=None, **kwargs):
         self.sslopts = ssl_opts if isinstance(ssl_opts, dict) else {}
-        self.sslopts["server_hostname"] = host
+        self._custom_endpoint = kwargs.get("custom_endpoint")
+        self._custom_port = kwargs.get("custom_port")
+        self.sslopts["server_hostname"] = self._custom_endpoint or host
         self._read_buffer = BytesIO()
-        super(SSLTransport, self).__init__(host, port=port, socket_timeout=socket_timeout, **kwargs)
+        super(SSLTransport, self).__init__(
+            self._custom_endpoint or host, port=self._custom_port or port, socket_timeout=socket_timeout, **kwargs
```

Comment:
can we swap the logic to have host & post checked first here ?

`host or self._custom_endpoint`

also wont host always have a value here ?

Reply:
host will always have a value but custom endpoint won't -- so CE or Host gives CE but None or Host gives host, if we swap the values it would always be host which we would not want

Reply:
let me check what uamqp did